### PR TITLE
Add threading to .all() and .filter() requests

### PR DESCRIPTION
**Pull request by tyler-8**
~~**WIP** do not merge yet~~

~~Opening this PR just for tracking purposes.~~

Fixes: https://github.com/digitalocean/pynetbox/issues/191

### CODE REVIEW

**Comment by tyler-8** [comment]
Build is failing because I'm using `concurrent.futures` which isn't available in Python 2. :\

**Comment by tyler-8** [comment]
This was a function to grab all the VLANs.

```python
nb = pynetbox.api(url, token=token)
nb2 = pynetbox.api(url, token=token, threading=True)

no_threading = timeit(nb)
# Took 0:01:18.703560
len(no_threading)
# 51900

threading = timeit(nb2)
# Took 0:00:27.203155
len(threading)
# 51900
```

File: pynetbox/core/query.py [hunk 1]
```diff
@@ -262,6 +270,24 @@ def _make_call(
...
+                new_params = {"offset": offset, "limit": page_size}
+                futures_to_results.append(
+                    pool.submit(self._make_call, add_params=new_params)
+                )
+
+            for future in cf.as_completed(futures_to_results):
+                try:
+                    result = future.result()
+                except Exception:
+                    # TODO: Raise exception with preferred method
```
Comment (tyler-8) [inline pynetbox/core/query.py]: I'd like to hear your thoughts on wait to do here @zachmoody - raise a `RequestError` and blow up the whole thing? Or perhaps a error log.

Comment (zachmoody) [reply pynetbox/core/query.py]: Admittedly haven't done a lot with thread, but what happens without the try-block?

Comment (tyler-8) [reply pynetbox/core/query.py]: Without the try block if the individual thread (one page of the API call) encountered an exception, that exception will be raised for the whole job.

Comment (zachmoody) [reply pynetbox/core/query.py]: Yeah, I think that's ok when the next best alternative (imo) is just catching and re-raising here. Kinda feel like getting an incomplete return should be terminal.

Comment (tyler-8) [reply pynetbox/core/query.py]: Done

**Comment by tyler-8** [comment]
I think I've got everything wrapped up on this for now.

**Comment by zachmoody** [comment]
Nice work! Will work on getting this merged and a new release cut today.

**Pull request merged**